File: sdks/go/voice.go
package pushfy

import (
	"context"
	"encoding/json"
)

// VoiceResource uploads audio and places voice calls.
type VoiceResource struct{ c *Client }

// VoiceUpload is a voice audio (.mp3) to upload. Filename defaults to
// "audio.mp3" when empty.
type VoiceUpload struct {
	Name     string
	Data     []byte
	Filename string
}

// UploadAudio uploads an audio file and returns the raw API result. The
// response does NOT return an audio id — the audio is stored under the Name
// you send here, so keep that exact name to place calls later with Send.
func (r *VoiceResource) UploadAudio(ctx context.Context, u VoiceUpload) (json.RawMessage, error) {
	filename := u.Filename
	if filename == "" {
		filename = "audio.mp3"
	}
	name := u.Name
	if name == "" {
		name = filename
	}
	form, err := buildMultipart(
		map[string]string{"nome": name},
		"audio", filename, "audio/mpeg", u.Data,
	)
	if err != nil {
		return nil, &InvalidRequestError{&PushfyError{Message: "build multipart: " + err.Error()}}
	}
	var out json.RawMessage
	err = r.c.classic(ctx, "POST", "/audio", classicOpts{form: form}, &out)
	return out, err
}

// VoiceCall places a call referencing a previously uploaded audio.
type VoiceCall struct {
	To string
	// AudioName is the audio's NAME — the exact nome you set when uploading via /audio.
	AudioName string
	ExtID     string
}

// Send places a voice call and returns the accepted messages.
func (r *VoiceResource) Send(ctx context.Context, call VoiceCall) ([]MessageResult, error) {
	msg := apiMessage{
		Destinations: []apiDest{{To: onlyDigits(call.To)}},
		Text:         "",
		ExtID:        call.ExtID,
		Audio:        call.AudioName,
	}
	var out []MessageResult
	err := r.c.classic(ctx, "POST", "/webapi", classicOpts{json: messagesEnvelope{Messages: []apiMessage{msg}}}, &out)
	return out, err
}

File: sdks/php/src/Resource/Voice.php
<?php

declare(strict_types=1);

namespace Pushfy\Resource;

use Pushfy\Pushfy;

/** Voice (text-to-call via pre-uploaded mp3 audio). */
final class Voice
{
    /** @var Pushfy */
    private $client;

    public function __construct(Pushfy $client)
    {
        $this->client = $client;
    }

    /**
     * Upload a voice audio (.mp3). Returns the API result.
     *
     * The response does NOT return an audio id — the audio is stored under the
     * `name` you send here, so keep that exact name to place calls later.
     *
     * @param array{name?: string, data: string, filename?: string} $params
     *        `data` is the raw mp3 bytes. `name` is the audio's name; retain it
     *        to reference the audio in `send`.
     * @return mixed
     */
    public function uploadAudio(array $params)
    {
        $filename = $params['filename'] ?? 'audio.mp3';
        $name = $params['name'] ?? $filename;
        $data = $params['data'] ?? '';

        return $this->client->classicRequest('POST', '/audio', [
            'multipart' => [
                ['name' => 'nome', 'contents' => $name],
                ['name' => 'audio', 'contents' => $data, 'filename' => $filename, 'contentType' => 'audio/mpeg'],
            ],
        ]);
    }

    /**
     * Place a voice call by referencing a previously uploaded audio.
     *
     * `audioName` is the audio's NAME — the exact `nome` you set when uploading via /audio.
     *
     * @param array{to: string, audioName: string, extId?: string} $params
     * @return mixed
     */
    public function send(array $params)
    {
        $msg = Message::normalize([
            'to' => $params['to'] ?? '',
            'text' => '',
            'extId' => $params['extId'] ?? null,
            'audio' => $params['audioName'] ?? null,
        ]);

        return $this->client->classicRequest('POST', '/webapi', [
            'json' => ['messages' => [$msg]],
        ]);
    }
}
